airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/test/pipeline.py

```diff
@@ -310,21 +311,39 @@ async def run_poe_tasks_for_package(
     airbyte_repo_dir = await get_filtered_airbyte_repo_dir(dagger_client, poetry_package_path)
     package_dir = await get_poetry_package_dir(airbyte_repo_dir, poetry_package_path)
     package_config = await get_airbyte_ci_package_config(package_dir)
-    container = prepare_container_for_poe_tasks(
-        dagger_client, airbyte_repo_dir, package_config, poetry_package_path, pipeline_context_params
-    )
+
     logger = logging.getLogger(str(poetry_package_path))
 
     if not package_config.poe_tasks:
         logger.warning("No poe tasks to run.")
         return []
 
-    poe_task_results = []
-    async with asyncer.create_task_group() as poe_tasks_task_group:
-        for task in package_config.poe_tasks:
-            poe_task_results.append(
-                poe_tasks_task_group.soonify(run_and_log_poe_task_results)(
-                    pipeline_context_params, str(poetry_package_path), container, task, logger
+    poe_task_results: List[asyncer.SoonValue] = []
+    return_results = []
+
+
+    if pipeline_context_params["python_version"]:
+        if pipeline_context_params["python_version"] not in package_config.python_versions:
+            logger.error(f"The python version {pipeline_context_params['python_version']} is not supported by the package.")
+            return []
+        package_config.python_versions = [pipeline_context_params["python_version"]]
+
+    if len(package_config.python_versions) > 1:
+        logger.info(f"Python versions: {package_config.python_versions}")
+
+    for python_version in package_config.python_versions:
+        container = prepare_container_for_poe_tasks(
+            dagger_client, airbyte_repo_dir, package_config, poetry_package_path, pipeline_context_params, python_version
+        )
+
+        async with asyncer.create_task_group() as poe_tasks_task_group:
+            for task in package_config.poe_tasks:
+                poe_task_results.append(
+                    poe_tasks_task_group.soonify(run_and_log_poe_task_results)(
```

Comment:
Let's pass the python version into `run_and_log_poe_task_results`, then you can use it on line 289 in `update_commit_check` with info on what python version that check is about — that would make it so we have a commit check per python version, I hope

Reply:
That's what we do at line `343` - we create a child logger that has a version as part of its name. If it's only a visual separation - it's there.

Reply:
the last argument of https://github.com/airbytehq/airbyte/pull/40731/files/9a0b66afd11cf43571e1ec03583834bf00f172cf#diff-b86e6d2f27d70812919004f8b786a5bb0ea5a7a28ec99c5ffd8773d9eb424eecR343

Reply:
`logger.getChild(f"@{python_version}"`